src/PrestaShopBundle/Bridge/Listener/InitControllerListener.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge\Listener;
+
+use Context;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\Localization\Locale\Repository;
+use PrestaShopBundle\Bridge\Controller\ControllerBridgeInterface;
+use PrestaShopBundle\Bridge\Controller\ControllerConfigurationFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+use Tab;
+use Tools;
+
+/**
+ * Init Controller by configuring something needs in all Controller migrate horizontally
+ */
+class InitControllerListener
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var ControllerConfigurationFactory
+     */
+    private $configurationFactory;
+
+    /**
+     * @var Repository
+     */
+    private $localeRepository;
+
+    public function __construct(LegacyContext $legacyContext, ControllerConfigurationFactory $configurationFactory, Repository $localeRepository)
+    {
+        $this->context = $legacyContext->getContext();
+        $this->configurationFactory = $configurationFactory;
+        $this->localeRepository = $localeRepository;
+    }
+
+    public function onKernelController(ControllerEvent $event)
+    {
+        if (!$this->supports($event)) {
+            return;
+        }
+
+        $controller = $event->getController()[0];
+
+        $controller->php_self = get_class($controller);
+        $this->context->controller = $controller;
+        $this->context->smarty->assign('link', $this->context->link);
+
+        $this->context->currentLocale = $this->localeRepository->getLocale(
+            $this->context->language->getLocale()
+        );
+
+        $controller->controllerConfiguration = $this->configurationFactory->create([
+            'id' => Tab::getIdFromClassName(
+                get_class($controller)::CONTROLLER_NAME_LEGACY
```

Comment:
I wonder if this could be picked up via the request attributes since `_legacy_controller` should be accessible in a Symfony route

Although even if we can I don't know if its the best idea? But we already rely on it on all the migrated controllers so maybe not such a bad idea